code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -191,3 +175,46 @@
 	desc = "A modified handcannon with a self-replicating reserve of decommissioned weaponized nanites. Spit shards of frozen angry robots into the bad guys. While it doesn't manipulate temperature in of itself, it does cause an internal explosion in anyone who is severely hot."
 	icon_state = "cryopistol"
 	ammo_type = list(/obj/item/ammo_casing/energy/nanite/cryo)
+
+// The Deltra!
+/obj/item/gun/energy/laser/deltra
+	name = "\improper Deltra HX62-S"
+	desc = "The Aussec Deltra is the pinnacle of projected kinetics technology. Capable of producing forcefield barriers to protect the wielder without compromising shooting capabilites, this weapon proves exceptionally useful in riot suppression, terrorist insurgencies and inquisitorial purges."
+	icon_state = "deltra"
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	lefthand_file = 'icons/mob/inhands/weapons/64x_guns_left.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/64x_guns_right.dmi'
+	worn_icon_state = "deltra"
+	inhand_icon_state = null
+	shaded_charge = TRUE
+	ammo_x_offset = 1
+	weapon_weight = WEAPON_HEAVY
+	w_class = WEIGHT_CLASS_HUGE
+	ammo_type = list(/obj/item/ammo_casing/energy/deltra)
+	var/obj/item/forcefield_projector/fast/forcebarrier //The forcefield projector that our weapon uses on right click. It charges and functions independent of the weapon.
+
+/obj/item/gun/energy/laser/deltra/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 1.2 SECONDS, TRUE, 0.3, 0.1)
+	forcebarrier = new /obj/item/forcefield_projector/fast(src)
+
+/obj/item/gun/energy/laser/deltra/examine(mob/user)
+	. = ..()
+	if(forcebarrier) //just in case something happens
+		. += span_notice("Right-click to use the forcefield projector.")
+		. += span_notice("Alt-click to drop the forcefield barriers.")
+		. += span_notice("It is currently sustaining [LAZYLEN(forcebarrier.current_fields)]/[forcebarrier.max_fields] fields, and it's [round((forcebarrier.shield_integrity/forcebarrier.max_shield_integrity)*100)]% charged.")
+
+/obj/item/gun/energy/laser/deltra/AltClick(mob/user)
```

Comment:
why not have this be when you right click the weapon?

Reply:
I literally didn't think that was a thing. Nice.